datasource/src/test/scala/quasar/physical/s3/S3DataSourceSpec.scala

```diff
@@ -107,38 +110,51 @@ class S3DataSourceSpec extends DatasourceSpec[IO, Stream[IO, ?]] {
     }
   }
 
+  "reading a non-existent file raises ResourceError.PathNotFound" >> {
+    val creds = EitherT.right[Throwable](credentials)
+    val ds = creds.flatMap(c => mkDatasource[G](S3JsonParsing.JsonArray, testBucket, c))
```

Comment:
I think you can just use `datasource` instead of `ds` in l.118 and remove l.114 and l.115?

Reply:
We could make assertions on actual JVM exceptions instead of using `EitherT`, but I chose to use `EitherT` since I'm more familiar with that.